backend/src/service/generic.ts

```diff
@@ -84,8 +84,27 @@ export function genericService<T>(
     }
   };
 
+  const getAll = async () =>{
+    try {
+      let query = `SELECT * FROM ${tableName} order by id;`;
+      console.log(query)
+      const result = await connection.query(query);
+      const entityList = [];
+      for(const obj of result.rows){
+        const newEntity = new ModelClass(obj) as Author | Book;
+        newEntity.validate();
+        entityList.push(newEntity);
+      };
+      return entityList;
```

Comment:
Is typescript okay with this? I assumed we would have to do `return entityList as T;`.
Probably will have to if you say this function returns T, like: `const getAll = async (): Promise<T> => {`

If it's not giving you typing errors, you can live it as is if you want, but personally I always like specifying the return type in functions, makes it easier for other people to read.